runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7107,6 +7107,41 @@ done:;
 		return rc;
 	}
 
+/*
+ * Determine the equality of two objects
+ * @param[in] lhs the lhs object of acmp bytecodes
+ * @param[in] rhs the rhs object of acmp bytecodes
+ * return true if they are equal and false otherwise
+ */
+	VMINLINE bool
+	acmp(j9object_t lhs, j9object_t rhs) 
+	{
+        bool acmpResult = true;
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+        if (rhs != lhs) {
+            /* two objects are not equal */
+            acmpResult = false;
+        } else if (NULL == lhs) {
+            /* two objects are null */
+            acmpResult = true;
+        } else {
+            /* at least one object is value type */
+            J9Class * lhsClass = J9OBJECT_CLAZZ(_currentThread, lhs);
+            if (J9_IS_J9CLASS_VALUETYPE(lhsClass)) {
+                acmpResult = false;
+            }
+        }
+#else /* J9VM_OPT_VALHALLA_VALUE_TYPES */
```

Comment:
the code in the else can simply do `return rhs == lhs`, but the VT code shuold have one return point